src/System.Private.ServiceModel/src/System/ServiceModel/Channels/RTHttpClientHandler.cs

```diff
@@ -0,0 +1,475 @@
+            }
+
+            ProcessResponse(response);
+            return response;
+        }
+
+        private void ProcessResponse(HttpResponseMessage response)
+        {
+            ProcessResponseCookies(response);
+        }
+
+        // Taken from System.Net.CookieModule.OnReceivedHeaders
+        private void ProcessResponseCookies(HttpResponseMessage response)
+        {
+            if (UseCookies)
+            {
+                IEnumerable<string> values;
+                if (response.Headers.TryGetValues(HttpKnownHeaderNames.SetCookie, out values))
+                {
+                    foreach (string cookieString in values)
+                    {
+                        if (!string.IsNullOrWhiteSpace(cookieString))
+                        {
+                            try
+                            {
+                                // Parse the cookies so that we can filter some of them out
+                                CookieContainer helper = new CookieContainer();
+                                helper.SetCookies(response.RequestMessage.RequestUri, cookieString);
+                                CookieCollection cookies = helper.GetCookies(response.RequestMessage.RequestUri);
+                                foreach (Cookie cookie in cookies)
+                                {
+                                    // We don't want to put HttpOnly cookies in the CookieContainer if the system
+                                    // doesn't support the RTHttpBaseProtocolFilter CookieUsageBehavior property.
+                                    // Prior to supporting that, the WinRT HttpClient could not turn off cookie
+                                    // processing. So, it would always be storing all cookies in its internal container.
+                                    // Putting HttpOnly cookies in the .NET CookieContainer would cause problems later
+                                    // when the .NET layer tried to add them on outgoing requests and conflicted with
+                                    // the WinRT nternal cookie processing.
+                                    //
+                                    // With support for WinRT CookieUsageBehavior, cookie processing is turned off
+                                    // within the WinRT layer. This allows us to process cookies using only the .NET
+                                    // layer. So, we need to add all applicable cookies that are received to the
+                                    // CookieContainer.
+                                    if (RTCookieUsageBehaviorSupported || !cookie.HttpOnly)
+                                    {
+                                        CookieContainer.Add(response.RequestMessage.RequestUri, cookie);
+                                    }
+                                }
+                            }
+                            catch (Exception)
+                            {
```

Comment:
It's hard to know what was copied from elsewhere, but this silent catch/ignore for all exception types seems dangerous.  If it is a pure duplicate of code and will be removed at some point, I guess it's okay.  But if this is our code to maintain, I'd like to know whether this catch can be constrained to only specific exception types.